File: Yona/Yona/Models/BrandResult.swift
//
//  BrandResult.swift
//  Yona
//
//  One result from the Brandfetch Brand Search API.
//

import Foundation

struct BrandResult: Identifiable, Decodable, Hashable {
    let name: String?
    let domain: String
    let icon: String?

    var id: String { domain }

    /// Brand name, or a capitalized fallback derived from the domain.
    var displayName: String {
        if let name, !name.isEmpty { return name }
        let main = domain.split(separator: ".").first.map(String.init) ?? domain
        return main.prefix(1).uppercased() + main.dropFirst()
    }
}

File: android/app/src/main/java/com/yona/app/core/BrandResult.kt
package com.yona.app.core

import kotlinx.serialization.Serializable

/** A Brandfetch Brand Search hit: a brand's name, domain, and icon URL. */
@Serializable
data class BrandResult(
    val name: String? = null,
    val domain: String,
    val icon: String? = null,
) {
    /** Brand name, or a capitalized fallback derived from the domain. */
    val displayName: String
        get() {
            if (!name.isNullOrEmpty()) return name
            val main = domain.substringBefore(".")
            return main.replaceFirstChar { it.uppercase() }
        }
}
